### ISSUE TITLE
Option on Repository Master to allways override changes.

### ISSUE THREAD

**Issue by imraro** [syncthing/syncthing#497]
SUBJ.

**Comment by khaffner** [comment]
I need exactly this option, to automatically override a non-master node. This was beautifully achieved by btsync with the "read-only folder". 
In my setup, I have a couple folders that I share with some friends, and I'm the only one managing the contents of those folders.

**Comment by AudriusButkevicius** [comment]
This doesn't work, as master is an individual setting on that machine, not a cluster wide thing. You can easily see two masters with this auto-override option enabled ping ponging each others changes until the end of time.

**Comment by khaffner** [comment]
I can't se any reason why 2 nodes would want to be master of the same folder. Nothing would get synced between those two, right?
I think there should be some option to override automatically. Or a single button that says "override all" with a popup warning. I'm the only master of 20+ folders in my cluster, I can't be the only one :)
Or is there perhaps some command I can run, or use the API?

**Comment by uok** [comment]
two nodes master: since a devices doesn't know about the master folder setting of other devices it is possible (accidently) to play ping pong, which is the reason for the manuel override button

I also hope that in the futre there will be an advanced setting to make it automatic override.

**Comment by AudriusButkevicius** [comment]
You can write a bit of bash to automate this, but the reason I feel this is not part of syncthing is because it's very easy for thingsto go wrong, and drive your network usage through the roof by accident.

**Comment by Improvised** [comment]
I've been using Syncthing to share my iTunes library with myself and my wife. So I have the same library intact on 3 systems. The thing is, I want my work system to be the master system. My wife has no need or desire to edit the library.

Nevertheless, every time someone on one of the remote systems closes iTunes, the library files get updated. Syncthing then sees a more recent version of the lib and then tells the master that it's out of sync. I then have to manually override 

Maybe I'm using the wrong program to accomplish this, but Syncthing has worked great. If I could just set it to auto-override, it would be perfect.

It does not seem like it would be a difficult setting to implement, and then all of us idiots who don't know how to use software correctly would shut-up =)

**Comment by bigbear2nd** [comment]
Hi
I would recommend to add the library to the ignored files.. So you dont have the trouble and you can also save bandwidth.. I guess you only need to sync the music files..

**Comment by Improvised** [comment]
That makes sense, but the only problem is that I actually do want the library changes from the master to propagate out to the clients, just not the reverse. In fact, all my music is placed on a different drive/sync and I added a second sync folder just for the library configuration files. That's the sync/folder I need to have auto override.

**Comment by imitation** [comment]
Sorry to bring up this ancient ticket again, but having this as "wontfix" just makes Syncthing unsuitable for so many use cases @calmh 
There are applications where you want exactly that - a folder that is always pushed to slaves without them being able to modify it. Think "backup" or "curated project assets".

**Comment by lkwg82** [comment]
this is sth Im working on. It can be solved with another folder type (internally). This will come later.

**Comment by AudriusButkevicius** [comment]
I think you should discuss on the forum what you are doing, before spending the effort and then getting shot down for doing something others disagree with. I don't think there is an easy way to make this authoritative which I guess is whay part of people are after. Furthermore I plan to make some changes to the code that is needed to implement what you are talking about, so it would be nice to be aware so that we don't make the work twice.

**Comment by lkwg82** [comment]
of course, before I can start the effort, it needed to be prepared a long time before even of thinking of that.

**Comment by 62mkv** [comment]
Hi. Am pretty much newcomer to the Syntching, but this one question appears on the Internet again and again, with a main theme being people asking aka complaining "we want this to work THIS way", and then @AudriusButkevicius explaining to them that they pretty much should accept the current behaviour and live with it. 

Let me add my 2 cents. We are running a system (I am not it's architect/developer; I just have a pretty bad luck to have it running after the nice guy who started it all left the company for good), that uses Syncthing master folders in order to deliver large files (like, CentOS packages, db dump and application content) on a multiple remote clients (aka kiosks) asynchronously and transparently. Everything works smooth, actually, UNLESS we have to re-install some of the kiosks. in this case, Syncthing knows that this node was already having the files, and as long as after installation it definitely is missing them, Syncthing believes that the node has had them deleted and VOILA - Master folder is Out of sync. I can of course press the button "Override" but that sometimes means wasting precious time, as today, when I entered office in the morning being pretty sure that kiosk is ready for work, but helas - it has not even started synchronization!

So, basically, I would like to know if there's an API: 
1) to poll if a folder is "Out of Sync", 
2) to "Override changes"

There was a question about this in the conversation above, but I don't see a reply (besides some vague mentioning of writing bash), thus asking again. Thanks a lot!

**Comment by calmh** [comment]
@62mkv if you reinstall a device and erase the synced files that were there, you should make sure to also erase the index database. Syncthing will then start afresh and you won't run into this issue.

**Comment by AudriusButkevicius** [comment]
And yes, there is a rest api for both

**Comment by 62mkv** [comment]
ICYMI: I guess it's `/rest/db/status` to poll for status, and `/rest/db/override` to override..

**Comment by 62mkv** [comment]
@calmh may I ask you, how to "erase index database" properly? What I've done: I've boot in single-mode and moved /root/.config/syncthing folder (which has index-v0.13.1db in it) 

afterwards I've boot the system as usual, but one of the folders still became "Out of sync" displaying 3 files on a main screen, but none on the "details" form. what have I done wrong ?

**Comment by AudriusButkevicius** [comment]
Please use forum for support

**Comment by mgcrea** [comment]
Just to add my 2cts, I'm hitting this as well. For a backup setup, quite often my folders are tagged as "Out of Sync", so I have to manually check it regularly otherwise my backup are just stuck. What's really annoying is that since I sync hundred of thousands of files, the web-ui takes between 5 to 10 minutes to properly update the status of theses folders and show the "Override sync" button (stuck on "Unknown).

It looks like there is a real need for this feature so please don't dismiss it. Having to manually verify that my backups are properly syncing is not really the UX I'm expecting for such a critical task.

**Comment by AudriusButkevicius** [comment]
The core problem in your case is thatyou are using syncthing as a backup tool, which I think it isn't.

Why can't you just rsync the files and be done with it?

**Comment by mgcrea** [comment]
@AudriusButkevicius rsync is a manual process, I want an always syncing solution (eg. replacing dropbox), and the ability to replicate the backup between cloud slaves without having to upload every file twice. It didn't sound like an unreasonable request for a tool that presents itself as a replacement for "proprietary sync and cloud services with something open, trustworthy and decentralized.".

**Comment by AudriusButkevicius** [comment]
Anyways, your problem only exists because you alter files on the other side. If you didn't, there wouldn't be an issue to start with

**Comment by mgcrea** [comment]
I don't ever touch/edit these remote files, never really understood why they were flagged as out of sync. Thought it was due to version not being exactly the same between nodes (eg. osx not being updated automatically). Anyway, I guess "I'm (just) holding it wrong".

**Comment by AudriusButkevicius** [comment]
Well thats a separate issue you should really look into oppose to going for the nuclear option and nuking everything without understanding what actually changed or more importantly why.

**Comment by jeek** [comment]
This, a million times, this. All I want is a fire-and-forget syncing solution to copy all files from A to B and override anything B tries to do to them. This is all over the place as a much-desired feature by many people. At this point, it looks like I may have to fork and maintain my own version of this program to get what should be basic functionality.

Thanks for being open source so forking is at least an option, but please seriously consider adding this so constant manual intervention isn't needed to "automatically" sync.

**Comment by canton7** [comment]
> At this point, it looks like I may have to fork and maintain my own version of this program to get what should be basic functionality.

By all means do some work on it, but consider adding to the existing effort rather than going off on your own. See https://github.com/syncthing/syncthing/pull/3780
